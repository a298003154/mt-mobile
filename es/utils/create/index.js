import { createBEM } from './bem';
import { createComponent } from './component';
import { createI18N } from './i18n';
export function createNamespace(name) {
  name = "mt-" + name;
  return [createComponent(name), createBEM(name), createI18N(name)];
}