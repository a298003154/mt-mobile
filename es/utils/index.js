import _extends from "@babel/runtime/helpers/esm/extends";
import Vue from 'vue';
import { camelize } from './format/string';
export var isServer = Vue.prototype.$isServer;
export function noop() {}
export function isDef(value) {
  return value !== undefined && value !== null;
}
export function isObj(x) {
  var type = typeof x;
  return x !== null && (type === 'object' || type === 'function');
}
export function isPlainObj(obj) {
  return Object.prototype.toString.call(obj) === '[object Object]';
}
export function get(object, path) {
  var keys = path.split('.');
  var result = object;
  keys.forEach(function (key) {
    result = isDef(result[key]) ? result[key] : '';
  });
  return result;
}
export { createNamespace } from './create/index';
export { addUnit } from './format/unit';

function stringifyArray(value) {
  var res = '';
  var stringified;

  for (var i = 0, l = value.length; i < l; i++) {
    stringified = stringifyClass(value[i]);

    if (isDef(stringified) && stringified !== '') {
      if (res) res += ' ';
      res += stringified;
    }
  }

  return res;
}

function stringifyObject(value) {
  var res = '';
  Object.keys(value).forEach(function (key) {
    if (value[key]) {
      if (res) res += ' ';
      res += key;
    }
  });
  return res;
}

function parseArray(value) {
  var res = {};
  value.forEach(function (item) {
    res = _extends({}, res, {}, parseClass(item));
  });
  return res;
}

function parseString(value) {
  var res = {};
  value.split(' ').forEach(function (c) {
    res[c.trim()] = true;
  });
  return res;
}

export function parseClass(value) {
  if (Array.isArray(value)) {
    return parseArray(value);
  }

  if (typeof value === 'string' && value.trim() !== '') {
    return parseString(value);
  }

  if (isPlainObj(value)) {
    return value;
  }
}
export function stringifyClass(value) {
  if (Array.isArray(value)) {
    return stringifyArray(value);
  }

  if (isObj(value)) {
    return stringifyObject(value);
  }

  if (typeof value === 'string') {
    return value;
  }
  /* istanbul ignore next */


  return '';
}

var parseStyleText = function parseStyleText(cssText, camel) {
  if (cssText === void 0) {
    cssText = '';
  }

  var res = {};
  var listDelimiter = /;(?![^(]*\))/g;
  var propertyDelimiter = /:(.+)/;
  cssText.split(listDelimiter).forEach(function (item) {
    if (item) {
      var tmp = item.split(propertyDelimiter);

      if (tmp.length > 1) {
        var k = camel ? camelize(tmp[0].trim()) : tmp[0].trim();
        res[k] = tmp[1].trim();
      }
    }
  });
  return res;
};

export function parseStyle(value) {
  if (isObj(value)) {
    return value;
  }

  if (typeof value === 'string') {
    return parseStyleText(value);
  }

  return {};
}