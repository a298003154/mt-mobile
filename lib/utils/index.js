"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.noop = noop;
exports.isDef = isDef;
exports.isObj = isObj;
exports.isPlainObj = isPlainObj;
exports.get = get;
exports.parseClass = parseClass;
exports.stringifyClass = stringifyClass;
exports.parseStyle = parseStyle;
exports.addUnit = exports.createNamespace = exports.isServer = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _vue = _interopRequireDefault(require("vue"));

var _string = require("./format/string");

var _index = require("./create/index");

exports.createNamespace = _index.createNamespace;

var _unit = require("./format/unit");

exports.addUnit = _unit.addUnit;
var isServer = _vue.default.prototype.$isServer;
exports.isServer = isServer;

function noop() {}

function isDef(value) {
  return value !== undefined && value !== null;
}

function isObj(x) {
  var type = typeof x;
  return x !== null && (type === 'object' || type === 'function');
}

function isPlainObj(obj) {
  return Object.prototype.toString.call(obj) === '[object Object]';
}

function get(object, path) {
  var keys = path.split('.');
  var result = object;
  keys.forEach(function (key) {
    result = isDef(result[key]) ? result[key] : '';
  });
  return result;
}

function stringifyArray(value) {
  var res = '';
  var stringified;

  for (var i = 0, l = value.length; i < l; i++) {
    stringified = stringifyClass(value[i]);

    if (isDef(stringified) && stringified !== '') {
      if (res) res += ' ';
      res += stringified;
    }
  }

  return res;
}

function stringifyObject(value) {
  var res = '';
  Object.keys(value).forEach(function (key) {
    if (value[key]) {
      if (res) res += ' ';
      res += key;
    }
  });
  return res;
}

function parseArray(value) {
  var res = {};
  value.forEach(function (item) {
    res = (0, _extends2.default)({}, res, {}, parseClass(item));
  });
  return res;
}

function parseString(value) {
  var res = {};
  value.split(' ').forEach(function (c) {
    res[c.trim()] = true;
  });
  return res;
}

function parseClass(value) {
  if (Array.isArray(value)) {
    return parseArray(value);
  }

  if (typeof value === 'string' && value.trim() !== '') {
    return parseString(value);
  }

  if (isPlainObj(value)) {
    return value;
  }
}

function stringifyClass(value) {
  if (Array.isArray(value)) {
    return stringifyArray(value);
  }

  if (isObj(value)) {
    return stringifyObject(value);
  }

  if (typeof value === 'string') {
    return value;
  }
  /* istanbul ignore next */


  return '';
}

var parseStyleText = function parseStyleText(cssText, camel) {
  if (cssText === void 0) {
    cssText = '';
  }

  var res = {};
  var listDelimiter = /;(?![^(]*\))/g;
  var propertyDelimiter = /:(.+)/;
  cssText.split(listDelimiter).forEach(function (item) {
    if (item) {
      var tmp = item.split(propertyDelimiter);

      if (tmp.length > 1) {
        var k = camel ? (0, _string.camelize)(tmp[0].trim()) : tmp[0].trim();
        res[k] = tmp[1].trim();
      }
    }
  });
  return res;
};

function parseStyle(value) {
  if (isObj(value)) {
    return value;
  }

  if (typeof value === 'string') {
    return parseStyleText(value);
  }

  return {};
}