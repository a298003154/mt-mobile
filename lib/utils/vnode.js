"use strict";

exports.__esModule = true;
exports.cloneVNode = cloneVNode;

function cloneVNode(vnode) {
  var cloned = new vnode.constructor(vnode.tag, vnode.data, vnode.children && vnode.children.slice(), vnode.text, vnode.elm, vnode.context, vnode.componentOptions, vnode.asyncFactory);
  cloned.ns = vnode.ns;
  cloned.isStatic = vnode.isStatic;
  cloned.key = vnode.key;
  cloned.isComment = vnode.isComment;
  cloned.fnContext = vnode.fnContext;
  cloned.fnOptions = vnode.fnOptions;
  cloned.fnScopeId = vnode.fnScopeId;
  cloned.asyncMeta = vnode.asyncMeta;
  cloned.isCloned = true;
  return cloned;
}