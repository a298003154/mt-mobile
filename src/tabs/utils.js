import { raf } from '../utils/dom/raf';

export function scrollLeftTo(el, to, duration) {
  let count = 0;
  const from = el.scrollLeft;
  const frames = duration === 0 ? 1 : Math.round((duration * 1000) / 16);

  function animate() {
    el.scrollLeft += (to - from) / frames;

    if (++count < frames) {
      raf(animate);
    }
  }

  animate();
}
