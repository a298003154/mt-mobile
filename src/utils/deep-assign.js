/* eslint-disable no-use-before-define */
import { isDef, isObj } from '.';

const { hasOwnProperty } = Object.prototype;

function assignKey(to, from, key) {
  const val = from[key];

  if (!isDef(val)) {
    return;
  }

  if (!hasOwnProperty.call(to, key) || !isObj(val) || typeof val === 'function') {
    to[key] = val;
  } else {
    to[key] = deepAssign(Object(to[key]), from[key]);
  }
}

export function deepAssign(to, from) {
  Object.keys(from).forEach((key) => {
    assignKey(to, from, key);
  });

  return to;
}
