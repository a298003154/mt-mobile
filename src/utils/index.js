import Vue from 'vue';

import { camelize } from './format/string';

export const isServer = Vue.prototype.$isServer;

export function noop() {}

export function isDef(value) {
  return value !== undefined && value !== null;
}

export function isObj(x) {
  const type = typeof x;
  return x !== null && (type === 'object' || type === 'function');
}

export function isPlainObj(obj) {
  return Object.prototype.toString.call(obj) === '[object Object]';
}

export function get(object, path) {
  const keys = path.split('.');
  let result = object;

  keys.forEach((key) => {
    result = isDef(result[key]) ? result[key] : '';
  });

  return result;
}

export { createNamespace } from './create/index';
export { addUnit } from './format/unit';

function stringifyArray(value) {
  let res = '';
  let stringified;
  for (let i = 0, l = value.length; i < l; i++) {
    stringified = stringifyClass(value[i]);
    if (isDef(stringified) && stringified !== '') {
      if (res) res += ' ';
      res += stringified;
    }
  }
  return res;
}

function stringifyObject(value) {
  let res = '';
  Object.keys(value).forEach((key) => {
    if (value[key]) {
      if (res) res += ' ';
      res += key;
    }
  });
  return res;
}

function parseArray(value) {
  let res = {};
  value.forEach((item) => {
    res = {
      ...res,
      ...parseClass(item)
    };
  });
  return res;
}

function parseString(value) {
  const res = {};
  value.split(' ').forEach((c) => { res[c.trim()] = true; });
  return res;
}

export function parseClass(value) {
  if (Array.isArray(value)) {
    return parseArray(value);
  }
  if (typeof value === 'string' && value.trim() !== '') {
    return parseString(value);
  }
  if (isPlainObj(value)) {
    return value;
  }
}

export function stringifyClass(value) {
  if (Array.isArray(value)) {
    return stringifyArray(value);
  }
  if (isObj(value)) {
    return stringifyObject(value);
  }
  if (typeof value === 'string') {
    return value;
  }
  /* istanbul ignore next */
  return '';
}

const parseStyleText = (cssText = '', camel) => {
  const res = {};
  const listDelimiter = /;(?![^(]*\))/g;
  const propertyDelimiter = /:(.+)/;
  cssText.split(listDelimiter).forEach((item) => {
    if (item) {
      const tmp = item.split(propertyDelimiter);
      if (tmp.length > 1) {
        const k = camel ? camelize(tmp[0].trim()) : tmp[0].trim();
        res[k] = tmp[1].trim();
      }
    }
  });
  return res;
};

export function parseStyle(value) {
  if (isObj(value)) {
    return value;
  }
  if (typeof value === 'string') {
    return parseStyleText(value);
  }
  return {};
}
