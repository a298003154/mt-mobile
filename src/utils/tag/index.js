export default {
  name: 'tag',
  props: {
    type: {
      type: String,
      default: 'div'
    }
  },
  render(h) {
    return (
      <this.type>
        {this.$slots.default}
      </this.type>
    );
  }
};
