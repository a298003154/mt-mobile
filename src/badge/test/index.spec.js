import { mount, later } from '../../../test/utils';
import Badge from '..';

test('callback events', () => {
  const callback = jest.fn();
  const onInput = jest.fn();
  const onCancel = jest.fn();
  const onSelect = jest.fn();

  const actions = [
    { name: 'Option', callback },
    { name: 'Option', disabled: true },
    { name: 'Option', loading: true },
    { name: 'Option', subname: 'Subname' }
  ];

  const wrapper = mount(Badge, {
    propsData: {
      value: true,
      actions,
      cancelText: 'Cancel'
    },
    listeners: {
      input: onInput,
      cancel: onCancel,
      select: onSelect
    }
  });

  const options = wrapper.findAll('.mt-action-sheet__item');
  options.at(0).trigger('click');
  options.at(1).trigger('click');
  wrapper.find('.mt-action-sheet__cancel').trigger('click');

  expect(callback).toHaveBeenCalled();
  expect(onCancel).toHaveBeenCalled();
  expect(onInput).toHaveBeenCalledWith(false);
  expect(onSelect).toHaveBeenCalledWith(actions[0], 0);
  expect(wrapper).toMatchSnapshot();
});

test('click overlay and close', async () => {
  const onInput = jest.fn();
  const onClickOverlay = jest.fn();
  const div = document.createElement('div');

  mount({
    template: `
      <div>
        <action-sheet
          :value="true"
          :get-container="getContainer"
          @input="onInput"
          @click-overlay="onClickOverlay"
        />
      </div>
    `,
    components: {
      Badge
    },
    data() {
      return {
        getContainer: () => div
      };
    },
    methods: {
      onInput,
      onClickOverlay
    }
  });

  await later();

  div.querySelector('.mt-overlay').click();
  expect(onInput).toHaveBeenCalledWith(false);
  expect(onClickOverlay).toHaveBeenCalledTimes(1);
});

test('disable lazy-render', () => {
  const wrapper = mount(Badge, {
    propsData: {
      lazyRender: false,
      actions: [
        { name: 'Option' },
        { name: 'Option' }
      ],
      cancelText: 'Cancel'
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test('render title and default slot', () => {
  const wrapper = mount(Badge, {
    propsData: {
      title: 'Title'
    },
    scopedSlots: {
      default() {
        return 'Default';
      }
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test('get container', () => {
  const wrapper = mount(Badge, {
    propsData: {
      value: true,
      getContainer: 'body'
    }
  });

  expect(wrapper.vm.$el.parentNode).toEqual(document.body);
});

test('close-on-click-action prop', () => {
  const onInput = jest.fn();
  const wrapper = mount(Badge, {
    propsData: {
      value: true,
      actions: [{ name: 'Option' }],
      closeOnClickAction: true
    },
    listeners: {
      input: onInput
    }
  });

  const option = wrapper.find('.mt-action-sheet__item');
  option.trigger('click');

  expect(onInput).toHaveBeenCalledWith(false);
});

test('round prop', () => {
  const wrapper = mount(Badge, {
    propsData: {
      value: true,
      round: true,
      actions: [{ name: 'Option' }]
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test('color option', () => {
  const wrapper = mount(Badge, {
    propsData: {
      value: true,
      actions: [{ name: 'Option', color: 'red' }]
    }
  });

  expect(wrapper).toMatchSnapshot();
});
