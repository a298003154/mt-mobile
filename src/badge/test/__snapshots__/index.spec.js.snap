// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['color option 1'] = '<span class="mt-badge mt-badge--not-a-wrapper" value="true" actions="[object Object]"> <!----></span>';

exports['disable lazy-render 1'] = '<span class="mt-badge mt-badge--not-a-wrapper" actions="[object Object],[object Object]" canceltext="Cancel"> <!----></span>';

exports['render title and default slot 1'] = '<span class="mt-badge mt-badge--not-a-wrapper" title="Title">Default <!----></span>';

exports['round prop 1'] = '<span class="mt-badge mt-badge--not-a-wrapper" value="true" round="true" actions="[object Object]"> <!----></span>';
